src/rust/analyzer-dispatcher/src/dispatch_event.rs

```diff
@@ -19,6 +22,33 @@ impl AnalyzerDispatchEvent {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AnalyzerDispatchEvents {
+    events: Vec<AnalyzerDispatchEvent>,
+}
+
+impl AnalyzerDispatchEvents {
+    pub fn new() -> Self {
+        Self { events: vec![] }
+    }
+}
+
+impl Default for AnalyzerDispatchEvents {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl From<Vec<AnalyzerDispatchEvent>> for AnalyzerDispatchEvents {
+    fn from(events: Vec<AnalyzerDispatchEvent>) -> Self {
+        Self { events }
+    }
+}
+
+impl ServiceMessage for AnalyzerDispatchEvents {
+    const TYPE_NAME: &'static str = "AnalyzerDispatchEvent";
+}
```

Comment:
There's a singular-plural mismatch here, is that expected?

Reply:
hmm I'll check, seems suspicious

Reply:
still there